backend/tracim_backend/lib/core/user.py

```diff
@@ -631,6 +723,14 @@ def _check_email_correctness(self, email: str) -> bool:
             return False
         return True
 
+    def _check_username_correctness(self, username: str) -> bool:
+        if len(username) < User.MIN_USERNAME_LENGTH or len(username) > User.MAX_USERNAME_LENGTH:
+            return False
+        for char in username:
+            if char not in "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_":
+                return False
+        return True
+
```

Comment:
Isnt't it the role of marshmallow schemas to do such checks ?

Reply:
It could be. I followed current code architecture. Do you think is preferable to change this ?

Reply:
Well it's not that important (meaning this code is working), but it would be better to use the right tools for the right thing. So I let you judge if this would be time-consuming to move those checks in the marshmallow schemas.

BTW python has a `string` module that defines the a-z/A-Z/0-9 strings : https://docs.python.org/3.7/library/string.html